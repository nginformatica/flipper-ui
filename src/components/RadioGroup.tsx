import {
    FormControl,
    FormControlLabel,
    FormLabel,
    Radio,
    RadioGroup as MuiRadioGroup
} from '@material-ui/core'
import React from 'react'

interface IProps {
    title?: string
    name: string
    value?: string
    options?: Array<{ value: string, label?: string }>
    style?: object
    onChange?: (event) => void
}

const RadioGroup = ({ options = [], style = {}, title, value, onChange }: IProps) =>
    <FormControl style={ style }>
        <FormLabel component='legend'>{ title }</FormLabel>
        <MuiRadioGroup
            name={ name }
            value={ value }
            onChange={ onChange }>
            {
                options.map(option =>
                    <FormControlLabel
                        key={ option.value }
                        label={ option.label }
                        value={ option.value }
                        control={ <Radio /> }
                    />
                )
            }
        </MuiRadioGroup>
    </FormControl>

export default RadioGroup
