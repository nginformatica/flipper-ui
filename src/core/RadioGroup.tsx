import {
    FormControl,
    FormControlLabel,
    FormLabel,
    Radio,
    RadioGroup as MuiRadioGroup
} from '@material-ui/core'
import React, { ChangeEvent, FC, ReactNode } from 'react'
import { DefaultProps } from './types'

interface RadioGroupProps extends DefaultProps {
    row?: boolean
    title?: ReactNode
    name: string
    value?: string
    disabled?: boolean
    spacing?: 'default' | 'equal'
    options?: IOption[]
    onChange?: (event: ChangeEvent<HTMLElement>) => void
}

interface IOption {
    value: string
    label?: ReactNode
    disabled?: boolean
}

const RadioGroup: FC<RadioGroupProps> = ({
    row,
    options = [],
    className,
    padding,
    margin,
    style = {},
    title,
    value,
    spacing,
    name,
    onChange,
    ...otherProps
}) =>
    <FormControl
        fullWidth
        className={ className }
        style={ { padding, margin, ...style } }
        { ...otherProps }>
        <FormLabel component='legend'>{ title }</FormLabel>
        <MuiRadioGroup
            row={ row }
            name={ name }
            value={ value }
            onChange={ onChange }>
            {
                options.map(option =>
                    <FormControlLabel
                        key={ option.value }
                        label={ option.label }
                        value={ option.value }
                        control={ <Radio /> }
                        disabled={ option.disabled }
                        style={ spacing === 'equal' ? { flex: 1 } : {} }
                    />
                )
            }
        </MuiRadioGroup>
    </FormControl>

export default RadioGroup
