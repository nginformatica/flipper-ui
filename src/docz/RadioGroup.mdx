---
name: RadioGroup
route: /radio-group
---

import RadioGroup from '../components/RadioGroup'
import { Playground, PropsTable } from 'docz'

# Picker

## Properties
<PropsTable of={ RadioGroup } />

## Demo
<Playground>
    <RadioGroup
        value="first"
        title="My options"
        name="options"
        options={
            [{
                label: 'First option',
                value: 'first'
            }, {
                label: 'Second Option',
                value: 'second'
            }]
        }
        onChange={ () => {} }
    />
</Playground>
