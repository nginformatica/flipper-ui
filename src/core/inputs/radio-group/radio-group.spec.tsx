import React from 'react'
import { render, screen } from '@testing-library/react'
import RadioGroup from '@/test/mocks/radio-group-mock'

describe('RadioGroup', () => {
    it('should render', () => {
        render(<RadioGroup />)
        const container = screen.getByTestId('radio-group-container')

        expect(container).toBeDefined()
    })

    it('should render with options', () => {
        render(
            <RadioGroup
                radioGroupProps={{
                    options: [
                        {
                            label: 'First Option',
                            value: 'first'
                        },
                        {
                            label: 'Second Option',
                            value: 'second'
                        }
                    ]
                }}
            />
        )

        const container = screen.getByTestId('radio-group-container')
        const firstOption = screen.getByText('First Option')
        const secondOption = screen.getByText('Second Option')

        expect(container).toBeDefined()
        expect(firstOption).toBeDefined()
        expect(secondOption).toBeDefined()
    })

    it('should render with spacing', () => {
        render(
            <RadioGroup
                radioGroupProps={{
                    spacing: 'equal',
                    options: [
                        {
                            label: 'First Option',
                            value: 'first'
                        },
                        {
                            label: 'Second Option',
                            value: 'second'
                        }
                    ]
                }}
            />
        )

        const container = screen.getByRole('radiogroup').firstElementChild

        expect(container).toHaveProperty('style.flex', '1 1 0%')
    })

    it('should match snapshot', () => {
        const { container } = render(
            <RadioGroup
                radioGroupProps={{
                    options: [
                        {
                            label: 'First Option',
                            value: 'first'
                        },
                        {
                            label: 'Second Option',
                            value: 'second'
                        }
                    ]
                }}
            />
        )

        expect(container).toMatchSnapshot()
    })
})
