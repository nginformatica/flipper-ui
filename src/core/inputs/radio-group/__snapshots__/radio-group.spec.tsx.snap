// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
    data-testid="radio-group-container"
  >
    <legend
      class="MuiFormLabel-root"
    >
      My options
    </legend>
    <div
      class="MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-16 Mui-checked MuiIconButton-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-18"
              name="options"
              type="radio"
              value="first"
            />
            <div
              class="PrivateRadioButtonIcon-root-19 PrivateRadioButtonIcon-checked-21"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-20"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          First Option
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-18"
              name="options"
              type="radio"
              value="second"
            />
            <div
              class="PrivateRadioButtonIcon-root-19"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-20"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Second Option
        </span>
      </label>
    </div>
  </div>
</div>
`;
